### ISSUE TITLE
Does a UI exist for Pelican?

### ISSUE THREAD

**Issue by patrickjohncollins** [getpelican/pelican#1601]
I'd like to know if a graphical user interface exists for Pelican?  It would be nice to be able to have a visual interface for creating a new site, setting the parameters, creating & editing articles, and more.  If one does not exist, I would be interested in creating one.  Comments?

**Comment by justinmayer** [comment]
That would be wonderful. The only related project I'm aware of is https://github.com/FedericoCeratto/shoebill by @FedericoCeratto.

**Comment by FedericoCeratto** [comment]
Thanks for mentioning it - is there any feature that you think Shoebill is lacking?

**Comment by patrickjohncollins** [comment]
Is there a demo of Shoebill ?  I'm not sure what it does.

**Comment by FedericoCeratto** [comment]
There is no demo, I've added few lines on the readme file.

**Comment by kernc** [comment]
Provided [pelican-md-yaml plugin](https://github.com/joachimneu/pelican-md-yaml) can be enabled, one can use [Prose](http://prose.io) to edit articles.

**Comment by jkw** [comment]
To do the core design you need to learn html/css/javascript and create your own [template](http://docs.getpelican.com/en/latest/themes.html). That code you can write in any editor that supports those languages.
For articles I don't know if you use .rst or .md, but there is a few nice Markdown editors out there. I use [Remarkable](http://remarkableapp.net/) in ArchLinux.

**Comment by patrickjohncollins** [comment]
Certainly, it is possible to use Pelican in combination with a text editor, personally I use Sublime Text on Mac OS X, and it works fine, but it is a rather manual process.  A graphical user interface would simplify greatly the tasks of creating a new site, setting up the parameters, selecting a pre-existing template, perhaps also providing the possibility to preview the site locally in a browser, and also publish the result to a server with the click of a button.  Such an interface would open the door to many users who are not as comfortable as we are at using the command line.

**Comment by FedericoCeratto** [comment]
@patrickjohncollins graphical instead of web based?

**Comment by patrickjohncollins** [comment]
@FedericoCeratto I think it would make most sense for the UI to be developed in HTML/CSS/JS so that it can be run on multiple platforms, either locally on the desktop or hosted on a server.

**Comment by jkw** [comment]
Yeah, that's true. It makes it more platform independant.

**Comment by kernc** [comment]
> it would make most sense for the UI to be developed in HTML/CSS/JS so that it can be run on multiple platforms, either locally on the desktop or hosted on a server.

Like [Prose](http://prose.io/)?

**Comment by SilverWingedSeraph** [comment]
I think a distinction should be made between a graphical text editor with Pelican support, and a Pelican management frontend. For example, a Prose-like editor could be deployed on a per-site basis, while a GTK- or Qt-based multisite management interface could be used by an administrator to back up, move, and deploy sites rapidly. Both would be extremely useful, however.

**Comment by iksteen** [comment]
I've been busy creating a (web based) IDE for Pelican based sites. It's very much a work in progress, but I've written a small post about it: http://blurringexistence.net/introducing-pelicide.html

There's also an interactive demo: http://blurringexistence.net/pelicide-demo/

[![Screenshot](http://blurringexistence.net/images/thumbnails/pelicide/384x0@100/pelicide-demo.png)](http://blurringexistence.net/images/pelicide/pelicide-demo.png)

**Comment by FedericoCeratto** [comment]
@iksteen nice UI - you might want to reuse code from https://github.com/FedericoCeratto/shoebill ... or maybe I might integrate your UI in it in future.

**Comment by DirkR** [comment]
It looks great!

**Comment by iksteen** [comment]
@FedericoCeratto Thanks! While that demo isn't connected to a back-end, but the back-end exists. It's basically a Twisted site that connects to a runner script which can perform various pelican tasks by importing the pelican modules and pelicanconf.py directly (it replaces the main pelican executable).

The code (both front-end and back-end) is available in a GitHub [repository](https://github.com/iksteen/pelicide).

@DirkR thanks!

**Comment by jonathan-s** [comment]
@iksteen very neat!

**Comment by leotrs** [comment]
If there are no plans to implement another pelican GUI, this should probably be closed.

**Comment by justinmayer** [comment]
For folks who want an administrative interface for Pelican, I suggest you encourage @tarekziade to expand upon his work on [Henet](https://github.com/AcrDijon/henet).

**Comment by dclong** [comment]
For command-line users, I have a cool [little blogging util](https://github.com/dclong/blog/blob/master/blog.py) written in Python which is based on Pelican. It supports adding, deleting, editing, searching (leveraging SQLite3 FTS5) blog posts, etc. It's coupled with my personal blog though but it might give people inspirations.

**Comment by FedericoCeratto** [comment]
@patrickjohncollins my UI https://github.com/FedericoCeratto/shoebill is indeed web-based. PRs are welcome.

**Comment by logological** [comment]
I haven't used it myself, but apparently [Netlify CMS](https://www.netlifycms.org/) works with Pelican, and even provides [a starter Pelican template](https://github.com/netlify-templates/pelican-netlify-cms).  If I understand correctly, Netlify CMS provides a web interface for editing statically generated sites.  It might not allow one to set up and modify the Pelican configuration, but I think it will allow one to add, change, and remove website content.

**Comment by GiovanH** [comment]
I'm not sure what the purpose of this issue is. Is it meant to track a new feature to create a UI as part of the main project, or a new subproject, or is it just an answered question?

**Comment by jkw** [comment]
As user states. GUI would be nice. As in app for Desktop. Linux/macOS/Windows that is. It's 2024 so anything else than platform independent is just stupid. 😏

**Comment by frica** [comment]
Hello,

This should be moved to a discussion, agree with @GiovanH.

It's an interesting topic though. In a time where the code editors are so versatile (I think of VS code but there are many others), can preview markdown, have a terminal view, what would be the use of that tool?

I can think a few features that would be interesting for me:
* quickly detect the drafts posts I have lingering around in my blog folder structure
* trigger with a shortcut/button the build & preview command `pelican && pelican -rl `
* autocomplete when I want to link an older post to the current one
* ...

I guess the feature list depends on your usage. 

Would be a nice side project, I would contribute if it'd use a modern python based UI framework like NiceGUI.
